server/BraveAlerterConfigurator.js

```diff
@@ -103,6 +104,28 @@ class BraveAlerterConfigurator {
 
         // If this is a OneSignal session (i.e. it isn't given the respondedByPhoneNumber) or if the SMS came from the session's respondedByPhoneNumber
         if (alertSession.respondedByPhoneNumber === undefined || alertSession.respondedByPhoneNumber === session.respondedByPhoneNumber) {
+          // Check for an invalid request to reset
```

Comment:
The tests in `test/unit/BraveAlerterConfiguratorTest/alertSessionChangedCalledbackTest.js` could be updated to include checks for your new logic.